testing_utilities/approximate_quantity_body.hpp

```diff
@@ -0,0 +1,161 @@
+}
+
+double ApproximateQuantity<double>::min() const {
+  return min_multiplier_;
+}
+
+double ApproximateQuantity<double>::max() const {
+  return max_multiplier_;
+}
+
+std::string ApproximateQuantity<double>::DebugString() const {
+  return "[" + quantities::DebugString(min_multiplier_) + ", " +
+         quantities::DebugString(max_multiplier_) + "]";
+}
+
+ApproximateQuantity<double>::ApproximateQuantity(
+    std::string const& representation,
+    double const min_multiplier,
+    double const max_multiplier)
+    : representation_(representation),
+      min_multiplier_(min_multiplier),
+      max_multiplier_(max_multiplier) {}
+
+template<typename Left, typename RDimensions>
+ApproximateQuantity<Product<Left, Quantity<RDimensions>>> operator*(
+    ApproximateQuantity<Left> const& left,
+    Quantity<RDimensions> const& right) {
+  return ApproximateQuantity<Product<Left, Quantity<RDimensions>>>(
+      left.representation_,
+      left.min_multiplier_,
+      left.max_multiplier_,
+      left.unit_ * right);
+}
+
+template<typename Left, typename RDimensions>
+ApproximateQuantity<Quotient<Left, Quantity<RDimensions>>> operator/(
+    ApproximateQuantity<Left> const& left,
+    Quantity<RDimensions> const& right) {
+  return ApproximateQuantity<Quotient<Left, Quantity<RDimensions>>>(
+      left.representation_,
+      left.min_multiplier_,
+      left.max_multiplier_,
+      left.unit_ / right);
+}
+
+template<typename Quantity>
+std::ostream& operator<<(std::ostream& out,
+                         ApproximateQuantity<Quantity> const& q) {
+  out << q.DebugString();
+  return out;
+}
+
+ApproximateQuantity<double> operator""_⑴(char const* const representation) {
+  return ApproximateQuantity<double>::Parse(representation, /*ulp=*/1);
+}
+
+ApproximateQuantity<double> operator""_⑵(char const* const representation) {
+  return ApproximateQuantity<double>::Parse(representation, /*ulp=*/2);
+}
+
+ApproximateQuantity<double> operator""_⑶(char const* const representation) {
+  return ApproximateQuantity<double>::Parse(representation, /*ulp=*/3);
+}
+
+ApproximateQuantity<double> operator""_⑷(char const* const representation) {
+  return ApproximateQuantity<double>::Parse(representation, /*ulp=*/4);
+}
+
+ApproximateQuantity<double> operator""_⑸(char const* const representation) {
+  return ApproximateQuantity<double>::Parse(representation, /*ulp=*/5);
+}
+
+ApproximateQuantity<double> operator""_⑹(char const* const representation) {
+  return ApproximateQuantity<double>::Parse(representation, /*ulp=*/6);
+}
+
+ApproximateQuantity<double> operator""_⑺(char const* const representation) {
+  return ApproximateQuantity<double>::Parse(representation, /*ulp=*/7);
+}
+
+ApproximateQuantity<double> operator""_⑻(char const* const representation) {
+  return ApproximateQuantity<double>::Parse(representation, /*ulp=*/8);
+}
+
+ApproximateQuantity<double> operator""_⑼(char const* const representation) {
```

Comment:
If we are going to do this beyond ⑴, add 🄐 .. 🄕 as well (morally they should `CHECK` for a hexadecimal `representation` but I am OK with trusting the sanity of the author here).

Maybe invest in a macro to factor the definitions?

Historical note: the parenthesized letters were encoded for compatibility with an ARIB standard https://www.unicode.org/wg2/docs/n3469.pdf (they are ARIB 9433..9458); some of the parenthesized digits are also present in that standard (ARIB 9417..9428), though they stop at 12 in ARIB (the parenthesized digits are from Unicode 1.1, so the rationale for their encoding may be lost to time).

Reply:
Done, with check and macro.